state/upgrades_test.go

```diff
@@ -2108,3 +2110,43 @@ func (s *upgradesSuite) TestAddRelationStatus(c *gc.C) {
 		expectUpgradedData{statuses, expectedStatuses},
 	)
 }
+
+func (s *upgradesSuite) TestCopyMongoSpaceToHASpaceConfigWhenValid(c *gc.C) {
+	c.Assert(getHASpaceConfig(s.state.db(), c), gc.Equals, "")
+
+	sn := network.SpaceName("mongo-space")
+	ms, err := s.state.SetOrGetMongoSpaceName(sn)
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(ms, gc.Equals, sn)
+
+	err = CopyMongoSpaceToHASpaceConfig(s.state)
+	c.Assert(err, jc.ErrorIsNil)
+
+	c.Check(getHASpaceConfig(s.state.db(), c), gc.Equals, "mongo-space")
+}
+
+func (s *upgradesSuite) TestNoCopyMongoSpaceToHASpaceConfigWhenNotValid(c *gc.C) {
+	c.Assert(getHASpaceConfig(s.state.db(), c), gc.Equals, "")
+
+	sn := network.SpaceName("mongo-space")
+	ms, err := s.state.SetOrGetMongoSpaceName(sn)
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(ms, gc.Equals, sn)
+	err = s.state.setMongoSpaceState(MongoSpaceInvalid)
+	c.Assert(err, jc.ErrorIsNil)
+
+	err = CopyMongoSpaceToHASpaceConfig(s.state)
+	c.Assert(err, jc.ErrorIsNil)
+
+	c.Check(getHASpaceConfig(s.state.db(), c), gc.Equals, "")
+}
+
+func getHASpaceConfig(db Database, c *gc.C) string {
```

Comment:
Ultimately, isn't this something that we should have on State anyway?

Reply:
Yes, there is actually no need to read it like this anyway. I will change it to use state.ControllerConfig(). As we only read it here.